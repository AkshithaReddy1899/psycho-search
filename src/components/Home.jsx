import React from 'react';

import MapComponent from './Map';

function Home() {
  return (
    <div className="px-auto">
      <div className="bg-no-repeat bg-cover bg-center bg-hero_sm md:bg-hero_md lg:bg-hero_lg p-4 lg:px-20">
        <div className="backdrop-blur-sm bg-white/10 grid justify-center items-center" style={{ height: '90vh', width: '90vw z-0' }}>
          <div className="px-12 md:px-20 lg:px-auto lg:w-1/2 lg:h-3/4 lg:pt-20 col-start-1 row-start-1 ">
            <div className="text-center md:text-left lg:h-3/4 lg:p-3 lg:pt-14 md:pt-0">
              <h1 className="text-5xl lg:text-7xl font-noto ">Welcome to PsychoSearch,</h1>
              <br className="h-10 pt-44" />
              <p className="text-lg lg:text-xl h-auto">
                The ultimate destination for those seeking information on the world&apos;s most notorious psychopaths. our website is the go-to resource for true crime enthusiasts and psychology students alike. Dive into our comprehensive profiles, packed with detailed biographies, crime scenes, and chilling quotes from the killers themselves. Stay informed and stay safe with PsychoSearch.
              </p>
            </div>
          </div>
        </div>
      </div>
      <h2 className="text-3xl p-4">Few of world&apos;s best known psychopaths</h2>
      <p className="text-slate-400 p-2">Click on the marker to know more about each of them</p>
      <MapComponent />
    </div>
  );
}

export default Home;
