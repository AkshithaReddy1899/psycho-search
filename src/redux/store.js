import {
  legacy_createStore as createStore,
  combineReducers,
  applyMiddleware,
  compose,
} from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import personsReducer from './person';

const rootReducer = combineReducers({
  person: personsReducer,
});

const store = createStore(rootReducer, compose(applyMiddleware(logger, thunk)));

export default store;
