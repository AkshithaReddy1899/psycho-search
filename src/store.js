import { configureStore } from '@reduxjs/toolkit';
import personSlice from './slices/personSlice';

const reducer = {
  person: personSlice,
};

const store = configureStore({
  reducer,
  devTools: true,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: false,
  }),
});

export default store;
