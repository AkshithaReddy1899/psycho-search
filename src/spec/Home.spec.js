import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import matches from './test-utils';
import store from '../redux/store';
import Home from '../components/Home';

describe('Home', () => {
  it('Home matched snapshot', () => {
    matches(<Provider store={store}><BrowserRouter><Home /></BrowserRouter></Provider>);
  });
});
