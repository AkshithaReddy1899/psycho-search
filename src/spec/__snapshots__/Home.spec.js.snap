// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Home matched snapshot 1`] = `
<div
  className="px-16 pt-6 pb-16"
>
  <div
    className="container text-center px-16 m-2"
  >
    <h3
      className="text-2xl p-2"
    >
      Fun facts about Psychopaths
    </h3>
    <div
      className="text-center w-2/5 p-8 mx-auto"
    >
      <ul
        className="text-left pb-12 text-lg m-auto place-content-center list-disc leading-7 px-6"
      >
        <li>
          Roughly one in every 200 of us is a psychopath.
        </li>
        <li>
          Psychopaths are people who do not recognize fear.
        </li>
        <li>
          People with psychopathy don't learn from their mistakes.
        </li>
        <li>
          Psychopaths don't respond to yawns well.
        </li>
        <li>
          People with psychopathic tendencies post tons of selfies.
        </li>
        <li>
          One of the classic traits of a psychopath is emotional poverty.
        </li>
        <li>
          Psychopaths are more likely to be night owls.
        </li>
      </ul>
    </div>
  </div>
  <h2
    className="text-3xl p-4"
  >
    Few of world's best known psychopaths
  </h2>
  <p
    className="text-slate-400 p-2"
  >
    Click on the marker to know more about each of them
  </p>
  <div
    className="border-2 mt-8 x-0 place-content-center"
    id="map"
    style={
      Object {
        "height": "650px",
        "margin": "auto",
        "width": "950px",
      }
    }
  >
    <p
      className="flex justify-center items-center m-auto mt-36"
    >
      <span
        style={
          Object {
            "display": "inherit",
            "height": "250px",
            "position": "relative",
            "width": "250px",
          }
        }
      >
        <span
          style={
            Object {
              "animation": "react-spinners-RingLoader-right 2s 0s infinite linear",
              "animationFillMode": "forwards",
              "border": "25px solid white",
              "borderRadius": "100%",
              "height": "250px",
              "left": "0",
              "opacity": "0.4",
              "perspective": "800px",
              "position": "absolute",
              "top": "0",
              "width": "250px",
            }
          }
        />
        <span
          style={
            Object {
              "animation": "react-spinners-RingLoader-left 2s 0s infinite linear",
              "animationFillMode": "forwards",
              "border": "25px solid white",
              "borderRadius": "100%",
              "height": "250px",
              "left": "0",
              "opacity": "0.4",
              "perspective": "800px",
              "position": "absolute",
              "top": "0",
              "width": "250px",
            }
          }
        />
      </span>
    </p>
  </div>
</div>
`;
